config/azure/build.yaml

```diff
@@ -43,15 +54,32 @@ jobs:
      displayName: 'Install Toolchain (Darwin)'
      condition: eq( variables['Agent.OS'], 'Darwin' )
 
+   - script: g++ --version
+     condition: or( eq( variables['Agent.OS'], 'Darwin' ), eq( variables['Agent.OS'], 'Linux' ) )
+   
+   # Windows needs the vcvarsall.bat command in the same script to retain environment variables for the scons task
    - script: |
-       g++ --version
+       call "C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Auxiliary\Build\vcvarsall.bat" x64
        scons -j ${{ parameters.threads }} install BUILD_TYPE=${{ parameters.buildType }} OPTIONS=${{ parameters.options }}
-     displayName: 'Build'
+     displayName: 'Build Windows'
+     condition: eq( variables['Agent.OS'], 'Windows_NT' )
+
+   - script: |
+       call "C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Auxiliary\Build\vcvarsall.bat" x64
+       scons testCore testCorePython testScene testImage testAlembic testAppleseed OPTIONS=${{ parameters.options }}
+       more test/IECore/results.txt
+     displayName: 'Test'
+     condition: eq( variables['Agent.OS'], 'Windows_NT' )
+
+   - script: scons -j ${{ parameters.threads }} install BUILD_TYPE=${{ parameters.buildType }} OPTIONS=${{ parameters.options }}
+     displayName: 'Build POSIX'
+     condition: or( eq( variables['Agent.OS'], 'Darwin' ), eq( variables['Agent.OS'], 'Linux' ) )
 
    - script: |
        scons testCore testCorePython testScene testImage testAlembic testAppleseed OPTIONS=${{ parameters.options }}
        cat test/IECore/results.txt
      displayName: 'Test'
+     condition: or( eq( variables['Agent.OS'], 'Darwin' ), eq( variables['Agent.OS'], 'Linux' ) )
```

Comment:
I don't know if they've improved this, but when we put the Gaffer pipeline together, it seemed that supplying a `condition` overrode the default behaviour of [continueOnError](https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=schema#job) (doh!) so I had to add `and( succeeded() ...` into all the ones that also required a platform switch to prevent the subsequent tasks running even when there was a build error.

https://github.com/GafferHQ/gaffer/blob/dd475cfc14b448696d32e8a104e2fe02bd814179/config/azure/build.yaml#L100

Reply:
Taken care of in 0fdd8a4